workflow/controller/workflowpod_test.go

```diff
@@ -69,6 +69,87 @@ func TestScriptTemplateWithVolume(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+var scriptTemplateWithArgsAndWithSource = `
```

Comment:
I thought I'd never say this, but there is a first time for everything.

I think you've written TOO many tests. 

I think you only need to test `extractMainCtrFromScriptTemplate.

Reply:
Haha no problem, I'll delete those that don't test `extractMainCtrFromScriptTemplate` explicitly. What I was trying to do is really test that exit code of a container in all these cases is zero, instead of just testing correct args order before execution.

Reply:
cool - I'll mark this as "Draft" while you do that

Reply:
Here you go, I left only 2 tests which cover `extractMainCtrFromScriptTemplate`. One for when script source is specified, and the other one when the source is empty.